### ISSUE TITLE
How to change the ConnectString programmatically?

### ISSUE THREAD

**Issue by kidmar** [ErikEJ/SqlCeToolbox#411]
Hi.
I just need to change the ConnectString to the SQLite Db so that I can specify the name of the file and a password at runtime.
Is it possible?
Thanks

**Comment by ErikEJ** [comment]
Sure, but the Toolbox is a design tool. Could you are more details?

**Comment by kidmar** [comment]
I've just build a very simple single table SQLite DB and I've generated the model and EF's stuff using your tool (and providing a ConnectString to connect to my file).

I've then used this code to insert a record on that DB:
> var db = new DbEntities();
>             
> if (db.MyTable.Count() == 0)
> {
> 	db.MyTable.Add(new MyTable());
> 	db.SaveChanges();
> }

I would like to create the db using something like:

> var db = new DbEntities(connectString);

The problem is that DbEntities does not provide that overload for DbContext.
Am I missing something or should I modify DBEntities class even though it was auto-generated?

**Comment by ErikEJ** [comment]
Is the DbEntities class not partial?

**Comment by kidmar** [comment]
Uh... Doh! You're right, I didn't notice that :(
Anyway I don't know how to generate a connection string that works with it...
I hit the UnintentionalCodeFirstException error and can't find a way to get past it
I've tried with a SQLiteConnection but it doesn't work.
I've also tried with System.Data.Entity.Core.EntityClient.EntityConnection with no luck.
I'm sorry I feel so dumb, can you help me?
Thanks

**Comment by kidmar** [comment]
I've made it.
This is what I have done:
I've added a reference to `System.Data.Entity` to the project.
Added the `using System.Data.EntityClient;`

Built the SQLite connection string:
`string connectionString = new SQLiteConnectionStringBuilder() { DataSource = dbFile, Password = password }.ConnectionString;`

then I've built the connection string using values I've found in the app.config file.
```
var ecsb = new EntityConnectionStringBuilder();
            ecsb.ProviderConnectionString = connectionString;
            ecsb.Provider = "System.Data.SQLite.EF6";
            ecsb.Metadata = "res://*/DbModel.csdl|res://*/DbModel.ssdl|res://*/DbModel.msl";
```

Then used it to connect:
`var db = new DbEntities(efcs);`

Thank you for the help
